src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2965,10 +3001,10 @@ private bool HasUnboxingConversion(TypeSymbol source, TypeSymbol destination, re
 
             if (HasUnboxingTypeParameterConversion(source, destination, ref useSiteDiagnostics))
             {
-                return true;
+                return ConversionKind.TypeParameterUnboxing;
```

Comment:
Simple obvious case.  If we have an unboxing type parameter conversion, then that's our conversion kind.